authorizer-app/src/app/components/rest-source-authorization/rest-source-user-list.component.ts

```diff
@@ -33,101 +46,104 @@ export class RestSourceUserListComponent implements OnInit, AfterViewInit {
     'actions'
   ];
 
+  @Input('project') set project(project: string) { this.project$.next(project); }
+  private project$ = new BehaviorSubject<string>('');
+
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
-
-  errorMessage: string;
-  restSourceUsers: RestSourceUser[];
-  restSourceProjects: RadarProject[];
-  selectedProject = '';
-  totalItems = 0;
+  updateTrigger = new BehaviorSubject<string>('init');
+  filterValue = new BehaviorSubject<string>('');
+  page = new BehaviorSubject<PageEvent>({
+    pageIndex: 0,
+    pageSize: 50,
+    length: 0,
+    previousPageIndex: 0,
+  });
 
   dataSource: MatTableDataSource<RestSourceUser>;
 
   constructor(
     private restSourceUserService: RestSourceUserService,
     public dialog: MatDialog,
-    private router: Router,
-    private activatedRoute: ActivatedRoute
   ) {}
 
   ngOnInit() {
-    this.selectedProject = this.activatedRoute.snapshot.queryParams.project;
-
-    this.loadAllRestSourceProjects();
-    this.dataSource = new MatTableDataSource(this.restSourceUsers);
-    this.dataSource.filterPredicate = function(data, filter: string): boolean {
-      return (
-        data.id.toLowerCase().includes(filter) ||
-        data.userId.toLowerCase().includes(filter) ||
-        data.externalId.toString().includes(filter)
-      );
-    };
-    this.onChangeProject(this.selectedProject);
+    this.dataSource = new MatTableDataSource([]);
   }
 
-  /**
-   * Set the paginator and sort after the view init since this component will
-   * be able to query its view for the initialized paginator and sort.
-   */
   ngAfterViewInit() {
     this.dataSource.sort = this.sort;
+    this.paginator.pageSize = 50;
+    this.paginator.page.subscribe(next => this.page.next(next));
+    this.subscribeToUsers();
   }
 
-  applyFilter(filterValue: string) {
-    filterValue = filterValue.trim(); // Remove whitespace
-    filterValue = filterValue.toLowerCase(); // Datasource defaults to lowercase matches
-    this.dataSource.filter = filterValue;
+  subscribeToUsers(): Subscription {
+    const filterInput = this.filterValue
+      .pipe(
+        debounceTime(150),
```

Comment:
Would it be possible to increase this by a bit so that there's not too many requests?

Reply:
hi, can you make the change?

Reply:
made the change